verilog/tools/kythe/indexing_facts_tree_extractor.cc

```diff
@@ -75,15 +77,47 @@ IndexingFactNode ExtractOneFile(absl::string_view content,
     for (const auto& message : syntax_error_messages) {
       std::cout << message << std::endl;
     }
-    exit_status = 1;
   }
-  parse_ok = status.ok();
 
   const auto& text_structure = analyzer.Data();
   const auto& syntax_tree = text_structure.SyntaxTree();
 
   return BuildIndexingFactsTree(syntax_tree, analyzer.Data().Contents(),
-                                filename);
+                                filename, file_list_facts_tree,
+                                extracted_files);
+}
+
+// Returns the path of the file relative to the file list.
+std::string GetPathRelativeToFileList(absl::string_view file_list_dir,
+                                      absl::string_view filename) {
+  return verible::file::JoinPath(file_list_dir, filename);
+}
+
+}  // namespace
+
+IndexingFactNode ExtractFiles(std::vector<std::string> ordered_file_list,
+                              int& exist_status,
+                              absl::string_view file_list_dir) {
+  // Create a node to hold the dirname of the ordered file list and group all
+  // the files and acts as a ordered file list of these files.
+  IndexingFactNode file_list_facts_tree(IndexingNodeData(
+      {Anchor(file_list_dir, 0, 0)}, IndexingFactType::kFileList));
+  std::set<std::string> extracted_files;
+
+  for (const auto filename : ordered_file_list) {
+    std::string content;
+    std::string file_path = GetPathRelativeToFileList(file_list_dir, filename);
+    if (!verible::file::GetContents(file_path, &content).ok()) {
+      LOG(INFO) << "Erro while reading file: " << file_path;
+      exist_status = 1;
```

Comment:
Might make sense to fully drop this status. The current code reports failure if any of the files error at reading. Is that intended?

Reply:
I suggested in a different comment to keep per-file information in an associative container/cache.  Yes, operation should still be robust enough to continue past file/IO errors, and just lose some facts/edges.  It's ok if the overall exit status is 0, as long as there is a channel to collect and report non-fatal errors.